mqtt/mqtt-bridge/src/settings.rs

```diff
@@ -43,21 +40,140 @@ impl Settings {
         config.try_into()
     }
 
-    pub fn nested_bridge(&self) -> &NestedBridgeSettings {
-        &self.nested_bridge
+    pub fn upstream(&self) -> Option<&ConnectionSettings> {
+        self.upstream.as_ref()
+    }
+
+    pub fn remotes(&self) -> &Vec<ConnectionSettings> {
+        &self.remotes
     }
 
-    pub fn subscriptions(&self) -> &Subscriptions {
+    pub fn messages(&self) -> &MessagesSettings {
+        &self.messages
+    }
+}
+
+impl<'de> serde::Deserialize<'de> for Settings {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        #[derive(Debug, serde_derive::Deserialize)]
+        struct Inner {
+            #[serde(flatten)]
+            nested_bridge: CredentialProviderSettings,
+
+            upstream: UpstreamSettings,
+
+            remotes: Vec<ConnectionSettings>,
+
+            messages: MessagesSettings,
+        }
+
+        let value: Inner = serde::Deserialize::deserialize(deserializer)?;
+
+        let upstream_connection_settings = match value.nested_bridge.gateway_hostname.clone() {
```

Comment:
As we discussed offline I did the deconstruct, I also had to call clone for gateway_hostname before map because otherwise I would get this error:   
```
81 | /             nested_bridge
82 | |                 .gateway_hostname
   | |_________________________________- value moved here
83 |                   //.clone()
84 |                   .map(|gateway_hostname| ConnectionSettings {
   |                        ^^^^^^^^^^^^^^^^^^ value used here after partial move
...
89 |                       credentials: Credentials::Provider(nested_bridge),
   |                                                          ------------- use occurs due to use in closure
   |
```